src/QueryReflection/QueryReflection.php

```diff
@@ -161,15 +161,17 @@ private function stringifyResult(Type $type): Type
 
     private function stringifyType(Type $type): Type
     {
+        $containsNull = TypeCombinator::containsNull($type);
+
         $numberType = new UnionType([new IntegerType(), new FloatType()]);
 
-        if ($numberType->isSuperTypeOf($type)->yes()) {
+        if ($numberType->isSuperTypeOf(TypeCombinator::removeNull($type))->yes()) {
```

Comment:
this fixed another bug: stringify did not work when the previous type contained null